### PR TITLE
[ML][Pipelines] support ignore files in additional includes

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

source:
```
additional_includes
  .amlignore
  hello.py
  world.py
```
hello.py is in `.amlignore`
result before this PR:
```
additional_includes
  hello.py/hello.py
  world.py/world.py # this is a bug brought in by [this commit](https://github.com/Azure/azure-sdk-for-python/pull/27960)
```
result after this PR:
```
additional_includes
  world.py
```

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py [hunk 1]
```diff
@@ -192,7 +187,13 @@ def resolve(self) -> None:
                 self._resolve_folder_to_compress(additional_include, Path(tmp_folder_path))
             else:
                 dst_path = (tmp_folder_path / src_path.name).resolve()
-                self._copy(src_path, dst_path, ignore_file=IgnoreFile() if self._is_artifact_includes else None)
+                self._copy(
+                    src_path,
+                    dst_path,
+                    # note that artifact additional includes file will skip ignore logic
+                    # TODO: check whether we should keep all files from artifact
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py:194]: @lalala123123 @wangchao1230 Could you please help confirm logic here? Is there any corresponding document?

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py:194]: suppose `x.py` is both in artifact, local folder and ignore file, what should we do with it?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py [hunk 2]
```diff
@@ -176,34 +173,55 @@ def resolve(self) -> None:
...
+                    additional_include,
+                    Path(tmp_folder_path),
+                    # TODO: seems it won't work as current ignore file implementation is based on absolute path
+                    ignore_file=ignore_file,
+                )
             else:
                 dst_path = (tmp_folder_path / src_path.name).resolve()
-                self._copy(src_path, dst_path, ignore_file=IgnoreFile() if self._is_artifact_includes else None)
+                # support ignore file in additional includes
+                self._copy(src_path, dst_path, ignore_file=get_ignore_file(src_path))
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py:199]: if src_path is_file, we will do abundant get_ignore_file call?

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py (hunk 2 above)
Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py:199]: concern on ignore file in sub folder

**Review by lalala123123 — APPROVED** [review]

**Pull request merged**